File: from-3201-to-3300/3217-delete-nodes-from-linked-list-present-in-array/solution.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func modifiedList(nums []int, head *ListNode) *ListNode {
    exclude := make(map[int]bool)
    for _, num := range nums {
        exclude[num] = true
    }
    dummy := &ListNode{-1, head}
    curr := dummy
    for head != nil {
        if !exclude[head.Val] {
            curr.Next = head
            curr = curr.Next
        }
        head = head.Next
    }
    curr.Next = nil
    return dummy.Next
}

File: from-3201-to-3300/3217-delete-nodes-from-linked-list-present-in-array/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        exclude = set(nums)
        dummy = ListNode(-1, head)
        curr = dummy
        while head:
            if head.val not in exclude:
                curr.next = head
                curr = curr.next
            head = head.next
        curr.next = None
        return dummy.next
        